packages/lib/src/components/UPI/UPI.tsx

```diff
@@ -0,0 +1,108 @@
+import { h, RefObject } from 'preact';
+import UIElement from '../UIElement';
+import UPIComponent from './components/UPIComponent';
+import CoreProvider from '../../core/Context/CoreProvider';
+import Await from '../internal/Await';
+import QRLoader from '../internal/QRLoader';
+import { UIElementStatus } from '../types';
+import { UpiPaymentData } from './types';
+
+/**
+ * 'upi' tx variant is the parent one.
+ * 'upi_collect' and 'upi_qr' are the sub variants which are submitted according to the shopper interaction.
+ */
+const UPI_COLLECT = 'upi_collect';
+const UPI_QR = 'upi_qr';
+
+class UPI extends UIElement {
+    public static type = 'upi';
+
+    constructor(props: any) {
```

Comment:
I know the UPI component has props, but could we have them here too?
I feel like this makes more sense since it's where it will interface with outside of the library

Reply:
There is actually no specific props for the UPI component.

This `props` variable is here as it needs to be passed when calling `super` inside the constructor, as the UiElement constructor expects that.

Reply:
I see, and then UIElement actually doesn't have a typed props too. Because otherwise we could just extend that.